### ISSUE TITLE
test_repair.sh encounters unexpected mismatch after many loops

### ISSUE THREAD

**Issue by leftwo** [oxidecomputer/crucible#468]
After many (like thousands) of test_repair.sh loops, I can get a failure post repair where crutest is expecting a different value for a block than the block contains.

Messages in the logs look like this:
```
Mismatch in Block:147 Volume offset:75265  Expected:48 Got:0
Update block 147 to 63048 (min:48 max:48 res:false cc[147]:63048 value:0)
Block 147 Also fails valid seed range
Mismatch in Block:147 Volume offset:75266  Expected:48 Got:0
Update block 147 to 63048 (min:48 max:48 res:false cc[147]:63048 value:0)
Block 147 Also fails valid seed range
```
Though, those messages are pretty generic and (if this is someone from the future) you are probably not seeing this issue based just on seeing those messages, as they will appear for almost any mismatch.

**Comment by leftwo** [comment]
Data gathering so far, the issue is always after repairing downstairs 0.

More data gathering, I can see this happens when
1) The u8 counter rolls over
2) The same block is written to twice in the same test loop.

I can also see from the logs that there is a problem in the update of the internal write counter when the verify step has determined that a block has the value in the correct range and is updating the index for that block:

```
Mismatch in Block:172 Volume offset:704513  Expected:0 Got:249
Update block 172 to 251 (min:248 max:0 res:true cc[172]:250 value:249)
Block 172 is in valid seed range
Mismatch in Block:172 Volume offset:704514  Expected:0 Got:249
Update block 172 to 252 (min:248 max:1 res:true cc[172]:251 value:249)
Block 172 is in valid seed range
Mismatch in Block:172 Volume offset:704515  Expected:0 Got:249
Update block 172 to 253 (min:248 max:2 res:true cc[172]:252 value:249)
Block 172 is in valid seed range
Mismatch in Block:172 Volume offset:704516  Expected:0 Got:249
Update block 172 to 254 (min:248 max:3 res:true cc[172]:253 value:249)
Block 172 is in valid seed range
Mismatch in Block:172 Volume offset:704517  Expected:0 Got:249
Update block 172 to 255 (min:248 max:4 res:true cc[172]:254 value:249)
Block 172 is in valid seed range
```

You can see that the `cc[172]` value keeps being incremented as it walks the bytes in a block.  It should not be continuing to update the counter once it does the first time through.  That's sure to be a problem.

**Comment by leftwo** [comment]
Fixed in 17367a721b03820dea7d08e9587d4eba4e76c785
